### PR TITLE
IItemHandler support

### LINKED ISSUE #75: IItemHandler capability support
Basically everything in [net.minecraftforge.items](https://github.com/PatchworkMC/YarnForge/tree/1.15.x/src/main/java/net/minecraftforge/items) as well as the relevant patches.

Required for [SBM Jukebox](https://www.curseforge.com/minecraft/mc-mods/sbm-jukebox)

### PR DESCRIPTION
**Pull request by famous1622**
Closes #75.

This PR adds IItemHandler capability support, as well as the ability for hoppers and droppers to interface with `IItemHandler` capability holders. I'll add support for the other way around (Forge mods interacting with vanilla and hopefully Fabric inventories) in another PR.

### CODE REVIEW

**Comment by kitlith** [comment]
I don't like currentSide, but I also don't have any great ideas on how to replace it, so *shrug*.

What's the scope of the patchwork-items module as opposed to the patchwork-extensions-items module?

**Comment by famous1622** [comment]
Patchwork items is the for  net.minecraftforge.items package, which is basically the entirety of making Forge IItemHandlers and having them able to interact with vanilla

**Comment by rikka0w0** [comment]
The `patchwork-extensions-items` package contains the `net.minecraftforge.common.extensions.IForgeItem` and its implementation. `IForgeItem` provides additional support for controlling the behavior of the item entity, and the interaction between the item and the world. In theory, IItemHandler does not need IForgeItem.

File: patchwork-items/src/main/java/net/patchworkmc/mixin/items/MixinHopperBlockEntity.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+		if (block.hasBlockEntity()) {
+			BlockEntity blockEntity = world.getBlockEntity(blockPos);
+
+			if (blockEntity != null) {
+				Direction direction = PatchworkItems.currentSide.get();
+
+				if (direction != null) {
+					LazyOptional<IItemHandler> capability = ((ICapabilityProvider) blockEntity).getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, direction.getOpposite());
+					PatchworkItems.currentSide.set(null);
```
Comment (TheGlitch76) [inline patchwork-items/src/main/java/net/patchworkmc/mixin/items/MixinHopperBlockEntity.java:73]: Turns out setting a ThreadLocal to null causes a memory leak.
```suggestion
					PatchworkItems.currentSide.remove();
```

Comment (rikka0w0) [reply patchwork-items/src/main/java/net/patchworkmc/mixin/items/MixinHopperBlockEntity.java:73]: Confirmed, setting the value to null will not remove the reference to the Thread from the internal map.

**Comment by TheGlitch76** [comment]
Please update for patchwork-api-base

File: patchwork-items/src/main/resources/fabric.mod.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "schemaVersion": 1,
+  "id": "patchwork-items",
+  "version": "${version}",
+  "name": "Patchwork Items",
+  "description": "A reimplementation of the Minecraft Forge item handling APIs.",
+  "authors": [
+    "coderbot"
```
Comment (TheGlitch76) [inline patchwork-items/src/main/resources/fabric.mod.json:8]:
```suggestion
    "PatchworkMC"
```

**Pull request merged**